Tests/Marketplace/search_and_uninstall_pack.py

```diff
@@ -18,6 +20,126 @@
+                                 method: str,
+                                 request_timeout: int | None = None,
+                                 accept: str = 'application/json',
+                                 attempts_count: int = 5,
+                                 sleep_interval: int = 60,
+                                 should_try_handler: Callable[[], bool] | None = None,
+                                 success_handler: Callable[[Any], Any] | None = None,
+                                 api_exception_handler: Callable[[ApiException], Any] | None = None,
+                                 http_exception_handler: Callable[[HTTPError | HTTPWarning], Any] | None = None):
+    """
+
+    Args:
+        client: demisto client.
+        retries_message: message to print after failure when we have more attempts.
+        exception_message: message to print when we get and exception that is not API or HTTP exception.
+        prior_message: message to print when a new retry is made.
+        path: endpoint to send request to.
+        method: HTTP method to use.
+        request_timeout: request param.
+        accept: request param.
+        attempts_count: number of total attempts made.
+        sleep_interval: sleep interval between attempts.
+        should_try_handler: a method to determine if we should send the next request.
+        success_handler: a method to run in case of successful request (according to the response status).
+        api_exception_handler: a method to run in case of api exception.
+        http_exception_handler: a method to run in case of http exception
+
+    Returns: True if the request succeeded and status in case of waiting_for_process_to_end
+
+    """
+    try:
+        for attempts_left in range(attempts_count - 1, -1, -1):
+            try:
+                if should_try_handler and not should_try_handler():
+                    # if the method exist and we should not try again.
+                    return True, None
+
+                # should_try_handler return True, we are trying to send request.
+                logging.info(f"{prior_message}, attempts_left: {attempts_count - attempts_left}/{attempts_count}")
+                response, status_code, headers = demisto_client.generic_request_func(client,
+                                                                                     path=path,
+                                                                                     method=method,
+                                                                                     accept=accept,
+                                                                                     _request_timeout=request_timeout)
+
+                if 200 <= status_code < 300 and status_code != 204:
+                    if success_handler:
+                        # We have a method to run as we were returned a success status code.
+                        return success_handler(response)
```

Comment:
correct me if I'm wrong, except here we always return `None` in the second item of the response tuple, no?

also, consider returning the response itself by default instead of the status bool, because then the only way to return the response data using this method is by providing `success_handler=lambda res: res`.
if you want additional information like status, a common practice is to do something like:
```
res, err = func()
```
where `err` is the error data or `None` if no error.